--- origsrc/recode-3.7-beta2/configure.ac	2008-03-09 19:45:35.000000000 -0500
+++ src/recode-3.7-beta2/configure.ac	2008-04-22 15:56:25.609375000 -0500
@@ -38,7 +38,7 @@
 AM_WITH_DMALLOC
 
 AM_GNU_GETTEXT(external)
-AM_GNU_GETTEXT_VERSION([0.16])
+AM_GNU_GETTEXT_VERSION([0.15])
 
 AH_BOTTOM([
 #include "cleaner.h"
--- origsrc/recode-3.7-beta2/src/Makefile.am	2008-03-09 19:45:35.000000000 -0500
+++ src/recode-3.7-beta2/src/Makefile.am	2008-04-22 16:22:45.578125000 -0500
@@ -51,7 +51,8 @@
 librecode_la_SOURCES = argmatch.c charname.c combine.c exitfail.c \
 fr-charname.c hash.c iconv.c localcharset.c names.c outer.c quotearg.c \
 recode.c request.c strip-pool.c task.c xalloc-die.c xmalloc.c $(C_STEPS)
-librecode_la_LDFLAGS = -version-info 0:0:0
+librecode_la_LIBADD = $(LTLIBINTL)
+librecode_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 # Relative to $(srcdir).
 ALAIN = $(KELD)
@@ -83,7 +84,7 @@
 	grep -v '^# *line [0-9]' merged.tm2 > $(srcdir)/merged.c
 	rm merged.tm1 merged.tm2
 
-$(srcdir)/recode.1: recode
+$(srcdir)/recode.1: recode$(EXEEXT)
 	@if test -r $@ && test ! -w $@; then \
 	  echo "WARNING: Page \`$@' read only, not updated"; \
 	elif $(PERL) $(top_srcdir)/doc/help2man \
--- origsrc/recode-3.7-beta2/src/argmatch.c	2008-03-09 19:45:35.000000000 -0500
+++ src/recode-3.7-beta2/src/argmatch.c	2008-04-22 16:30:42.531250000 -0500
@@ -36,7 +36,6 @@
 # define _(Text) Text
 #endif
 
-#include "error.h"
 #include "quotearg.h"
 
 /* When reporting an invalid argument, show nonprinting characters
@@ -156,10 +155,10 @@
 argmatch_invalid (const char *context, const char *value, int problem)
 {
   char const *format = (problem == -1
-			? _("invalid argument %s for `%s'")
-			: _("ambiguous argument %s for `%s'"));
+			? _("librecode: invalid argument %s for `%s'")
+			: _("librecode: ambiguous argument %s for `%s'"));
 
-  error (0, 0, format, quotearg_style (ARGMATCH_QUOTING_STYLE, value), context);
+  fprintf (stderr, format, quotearg_style (ARGMATCH_QUOTING_STYLE, value), context);
 }
 
 /* List the valid arguments for argmatch.
--- origsrc/recode-3.7-beta2/src/xalloc-die.c	2008-03-09 19:45:35.000000000 -0500
+++ src/recode-3.7-beta2/src/xalloc-die.c	2008-04-22 16:36:10.781250000 -0500
@@ -23,9 +23,9 @@
 
 #include "xalloc.h"
 
+#include <stdio.h>
 #include <stdlib.h>
 
-#include "error.h"
 #include "exitfail.h"
 
 #include "gettext.h"
@@ -35,7 +35,8 @@
 void
 xalloc_die (void)
 {
-  error (exit_failure, 0, "%s", _("memory exhausted"));
+  fprintf (stderr, "librecode: %s", _("memory exhausted"));
+  exit (exit_failure);
 
   /* The `noreturn' cannot be given to error, since it may return if
      its first argument is 0.  To help compilers understand the
--- origsrc/recode-3.7-beta2/tests/setup.py.in	2008-03-09 19:45:35.000000000 -0500
+++ src/recode-3.7-beta2/tests/setup.py.in	2008-04-22 16:41:26.718750000 -0500
@@ -17,7 +17,8 @@
 
     setup(ext_modules=[
         Extension('Recode', ['Recode.c'],
-                  include_dirs=[top_builddir, os.path.join(top_srcdir, 'src')],
+                  include_dirs=[top_builddir, os.path.join(top_srcdir, 'lib'), os.path.join(top_srcdir, 'src')],
+                  library_dirs=[os.path.join(top_builddir, 'src', '.libs')],
                   libraries=['recode'])
         ])
 
